jbmc/src/java_bytecode/java_static_initializers.cpp

```diff
@@ -205,9 +208,43 @@ static void clinit_wrapper_do_recursive_calls(
   auto find_sym_it = symbol_table.symbols.find(real_clinit_name);
   if(find_sym_it != symbol_table.symbols.end())
   {
-    code_function_callt call_real_init;
-    call_real_init.function() = find_sym_it->second.symbol_expr();
-    init_body.move_to_operands(call_real_init);
+    if(!nondet_static)
+    {
+      code_function_callt call_real_init;
+      call_real_init.function() = find_sym_it->second.symbol_expr();
+      init_body.move_to_operands(call_real_init);
+    }
+    else
+    {
+      object_factory_parameterst parameters = object_factory_parameters;
+      parameters.function_id = real_clinit_name;
+
+      for(const auto &symbol : symbol_table.symbols)
+      {
+        if(
+          symbol.second.is_static_lifetime &&
+          symbol.second.type.get(ID_C_class) == class_name)
+        {
+          const symbol_exprt new_global_symbol = symbol.second.symbol_expr();
+
+          parameters.max_nonnull_tree_depth =
+            is_non_null_library_global(symbol.first)
+              ? object_factory_parameters.max_nonnull_tree_depth + 1
```

Comment:
Should this just be `std::max(1, object_factory_parameters.max_nonnull_tree_depth)`?